File: client/src/models/campaign/InitiativeSlot.ts
import {GenesysSymbols} from "../roll/GenesysSymbols";
import Character from "../actor/Character";

export default interface InitiativeSlot {
    type: Type
    results: Record<GenesysSymbols, number>
    character: Character
}

export enum Type {
    Player = 'Player',
    NPC = 'NPC'
}

File: server/src/main/java/com/github/genraven/genesys/domain/campaign/encounter/InitiativeSlot.java
package com.github.genraven.genesys.domain.campaign.encounter;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

@Data
public class InitiativeSlot {

    private Type type;
    private Character character;

    @Getter
    @AllArgsConstructor
    public enum Type {
        PLAYER("Player"),
        NPC("NPC");

        @JsonValue
        private final String label;
    }
}
